arch/arm/core/cortex_m/mpu/arm_core_mpu.c

```diff
@@ -77,8 +77,15 @@ void _arch_configure_static_mpu_regions(void)
 		.start = (u32_t)&_nocache_ram_start,
 		.size = (u32_t)&_nocache_ram_size,
 		.attr = K_MEM_PARTITION_P_RW_U_NA_NOCACHE,
-		}
+		},
 #endif /* CONFIG_NOCACHE_MEMORY */
+#if defined(CONFIG_ARCH_HAS_RAMFUNC_SUPPORT)
+		{
+		.start = (u32_t)&_ramfunc_ram_start,
+		.size = (u32_t)&_ramfunc_ram_size,
+		.attr = K_MEM_PARTITION_P_RX_U_RX,
```

Comment:
Not sure about un-privileged execution, here. This is a static region, perhaps we should align with the other static regions that are NA for nPRIV. @andrewboie what is your take?

(On the other hand, the entire flash on XIP images is open for execution for both priv/unpriv). But that's application and system code. In these RAMFUNC regions, do we intend to give them out "as-is" to user threads?

Reply:
For me the goal of `__ramfunc` is to be able to place and execute some code in RAM. It can be because the flash is temporarily unavailable (ie during flashing) or for performances reason. At least in the later case we probably want to be able to do that from user threads.

I therefore think that those regions should be aligned to the flash attributes like in this PR. That's why they are also not writable. Note that in practice they slightly differ on the cacheability (write through vs write back), but given the permissions, that doesn't make any difference.

Reply:
Alright

Reply:
We currently don't distinguish between kernel and user text areas, I'm OK with this policy.